src/main/java/org/terracotta/offheapstore/paging/UpfrontAllocatingPageSource.java

```diff
@@ -440,67 +473,146 @@ public synchronized Runnable removeAllocationThreshold(ThresholdDirection direct
       throw new AssertionError();
     }
 
-    private static Collection<ByteBuffer> allocateBackingBuffers(BufferSource source, long max, int maxChunk, int minChunk, boolean fixed) {
-      Collection<ByteBuffer> buffers = new LinkedList<ByteBuffer>();
+  /**
+   * Allocate multiple buffers to fulfill the requested memory {@code toAllocate}. We first divide {@code toAllocate} in
+   * chunks of size {@maxChunk} and try to allocate them in parallel on all available processors. If one chunk fails to be
+   * allocated, we try to allocate two chunks of {@code maxChunk / 2}. If this allocation fails, we continue dividing until
+   * we reach of size of {@code minChunk}. If at that moment, the allocation still fails, an {@code IllegalArgumentException}
+   * is thrown.
+   * <p>
+   * When {@code fixed} is requested, we will only allocated buffers of {@code maxChunk} size. If allocation fails, an
+   * {@code IllegalArgumentException} is thrown without any division.
+   * <p>
+   * If the allocation is interrupted, the method will exit and return what was allocated so far. The interrupt flag is
+   * set.
+   *
+   * @param source source used to allocate memory buffers
+   * @param toAllocate total amount of memory to allocate
+   * @param maxChunk maximum size of a buffer. This is the targeted size for all buffers if everything goes well
+   * @param minChunk minimum buffer size allowed
+   * @param fixed if all buffers should have a the same size (except the last one with {@code toAllocate % maxChunk != 0}, if true, {@code minChunk} isn't used
+   * @return the list of allocated buffers
+   * @throws IllegalArgumentException when we fail to allocate the requested memory
+   */
+    private static Collection<ByteBuffer> allocateBackingBuffers(final BufferSource source, long toAllocate, int maxChunk, final int minChunk, final boolean fixed) {
+
+      final long start = (LOGGER.isInfoEnabled() ? System.nanoTime() : 0);
```

Comment:
Good point. I was trying to avoid initializing `start` when not needed. It means that if someone changes the level in the middle of start and stop, he will get a really weird duration. Does it matter?